src/main/java/io/stargate/sgv2/jsonapi/api/model/command/clause/sort/SortClause.java

```diff
@@ -31,4 +31,13 @@ public boolean hasVsearchClause() {
             .path()
             .equals(DocumentConstants.Fields.VECTOR_EMBEDDING_FIELD);
   }
+
+  public boolean hasVectorizeSearchClause() {
+    return sortExpressions != null
+        && sortExpressions.stream()
+            .findFirst()
```

Comment:
Is it guaranteed that we can only have one sort expression with vectors? (I assume so based on code).
Maybe add a comment here stating that invariant.

Reply:
Yes, check for this is in SortClauseDeserializer.java where validation will throw VECTOR_SEARCH_USAGE_ERROR if it has more than one expression used.